util/unifyfs/src/unifyfs-rm.c

```diff
@@ -483,58 +485,120 @@ static int slurm_read_resource(unifyfs_resource_t* resource)
     char number[16];
 
-    if (server_argv != NULL) {
-        if (args->server_path != NULL) {
-            server_argv[0] = strdup(args->server_path);
-        } else {
-            server_argv[0] = strdup(BINDIR "/unifyfsd");
-        }
-    }
-    argc = 1;
-
-    if (args->debug) {
+    switch (args->arg_type) {
+    case UNIFYFS_CONSTRUCT_ARGS_UNIFYFSD:
         if (server_argv != NULL) {
-            server_argv[argc] = strdup("-v");
-            snprintf(number, sizeof(number), "%d", args->debug);
-            server_argv[argc + 1] = strdup(number);
+            if (args->server_path != NULL) {
+                server_argv[0] = strdup(args->server_path);
+            } else {
+                server_argv[0] = strdup(BINDIR "/unifyfsd");
+            }
         }
-        argc += 2;
-    }
+        argc = 1;
 
-    if (args->cleanup) {
-        if (server_argv != NULL) {
-            server_argv[argc] = strdup("-C");
-        }
-        argc++;
-    }
+	if (args->debug) {
+            if (server_argv != NULL) {
+	        server_argv[argc] = strdup("-v");
+		snprintf(number, sizeof(number), "%d", args->debug);
+		server_argv[argc + 1] = strdup(number);
+            }
+	    argc += 2;
+	}
 
-    if (args->consistency != UNIFYFS_CM_LAMINATED) {
+	if (args->cleanup) {
+            if (server_argv != NULL) {
+	        server_argv[argc] = strdup("-C");
+            }
+	    argc++;
+	}
+
+	if (args->consistency != UNIFYFS_CM_LAMINATED) {
+            if (server_argv != NULL) {
+	        server_argv[argc] = strdup("-c");
+		server_argv[argc + 1] =
+		    strdup(unifyfs_cm_enum_str(args->consistency));
+            }
+	    argc += 2;
+	}
+
+	if (args->mountpoint != NULL) {
+            if (server_argv != NULL) {
+	        server_argv[argc] = strdup("-m");
+                server_argv[argc + 1] = strdup(args->mountpoint);
+            }
+	    argc += 2;
+	}
+
+	if (server_argv != NULL) {
+	    server_argv[argc] = strdup("-S");
+	    server_argv[argc + 1] = strdup(args->share_dir);
+	    server_argv[argc + 2] = strdup("-H");
+	    server_argv[argc + 3] = strdup(args->share_hostfile);
+	    argc += 4;
+	}
+	break;
+    case UNIFYFS_CONSTRUCT_ARGS_STAGE_IN:
         if (server_argv != NULL) {
-            server_argv[argc] = strdup("-c");
-            server_argv[argc + 1] =
-                strdup(unifyfs_cm_enum_str(args->consistency));
-        }
-        argc += 2;
-    }
+            if (args->transfer_exe == NULL) {
+	        fprintf(stderr,
+			"%s: args->transfer_exe is NULL!\n",
+			__func__);
+                return -1;
+            }
+            if (args->stage_in == NULL) {
+                fprintf(stderr, "%s: args->stage_in is NULL!\n",
+			__func__);
+	        return -1;
+            }
 
-    if (args->mountpoint != NULL) {
+	    //	    if (args->server_path != NULL) {
+	    server_argv[0] = strdup(args->transfer_exe);
+	    //    } else {
+	    //	        server_argv[0] = strdup(BINDIR "/unifyfsd");
+	    //	    }
```

Comment:
Remove debug comments

Reply:
These aren't debug comments.  They're how the code will work once the stage-in/stage-out infrastructure is stable enough that we can reasonably count on a transfer-stage binary being in the right place.  Admittedly I didn't change the names in them for this, which I should have.  

My next commit has the names change as to how it will be implemented, and has explanatory accompanying comments explaining what it is and why it's there.